Automattic/jetpack
# JPOalypso: onboard connections

#### Changes proposed in this Pull Request:

* this, in conjunction with https://github.com/Automattic/wp-calypso/pull/17094 will onboard connections
* the provision script will set the Jetpack option `onboarding` to a token, the provision cancel script will clean it
* if a site has the Jetpack option `onboarding` set to a token, it will have a parameter `onboarding=TOKEN` in the connection url
* adds new test for onboarded connection
* accept `onboarding` parameter that will contain choices made by user during onboarding. It will be parsed and different tasks performed based on the choices, like updating the site title or creating a page and setting it as front page.

#### Testing instructions:

1. in your wpcom sandbox, apply D7342
2. locally, build and run https://github.com/Automattic/wp-calypso/pull/17094
3. switch to this branch in your sandboxorg
4. do `wp eval "Jetpack::create_onboarding_token();"`
5. to point this to your Calypso development environment, edit `class.jetpack.php` and uncomment `//$url = add_query_arg( 'calypso_env', 'development', $url );`
6. go to the WP Admin of this sandboxorg, to the Jetpack settings, connect it and you should be sent to the `/start/jetpack-onboarding` flow.
7. test only the first step of the onboarding flow, saving a Site Title and description and make sure it's saved back to the Jetpack site.
